app/v2/controllers/RequestContext.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.controllers
```

Comment:
Is there a reason we have not included an `app.api` package here for all shared functionality?

Reply:
It's because there are two version packages. I did start to pull out the latest v2 utility classes using them for v1 and v2 but it soon got messy (and risked breaking things -- e.g. error messages which we don't always test -- for v1) and started to be a distraction from the controller simplification itself. 

I also considered moving only the v2 utility classes to a package that looks like it is common. But then those classes are used only for v2 -- and so this could be confusing.

It feels like if this needs to be done, it should be done as a separate PR. And that would be easiest done once the v1 packages are deleted.